ocaml/xapi/helpers.ml

```diff
@@ -1184,6 +1184,17 @@ let retry ~__context ~doc ?(policy = Policy.standard) f =
 let retry_with_global_lock ~__context ~doc ?policy f =
   retry ~__context ~doc ?policy (fun () -> with_global_lock f)
 
+(* Retry function f until success or timeout, f should return true on success *)
+let rec retry_until_timeout ?(interval=0.1) ?(timeout=5.) doc f =
+  match f() with
+  | true -> ()
+  | false -> let next_interval = interval *. 1.5 in
+    let next_timeout = timeout -. interval in
+    if next_timeout < 0. then
+      raise Api_errors.(Server_error(internal_error, [Printf.sprintf "retry %s failed" doc]));
+    Thread.delay interval;
```

Comment:
This needs to use `next_interval`. It would be easier to just say `let interval = interval *. 1.5` above.

Reply:
No, this should be `interval` 
as `let next_timeout = timeout -. interval`,
I changed the code and it becomes a little more complicated to understand.
The change is basing on following thinking
If we sleep, we should try once more after wake up. (previous version may sleep, and raise error after wait up other than retry)

Reply:
> This needs to use `next_interval`. It would be easier to just say `let interval = interval *. 1.5` above.

No, this should be interval
as let next_timeout = timeout -. interval,
I changed the code and it becomes a little more complicated to understand.
The change is basing on following thinking
If we sleep, we should try once more after wake up. (previous version may sleep, and raise error after wait up other than retry)

Reply:
Ok - so `Thread.delay` uses the old interval and the next iteration will use the next interval.